File: src/clients/CSharp/TakeIoLib/Entities/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TakeIoLib.Entities
{
    using System;
    using System.Collections.Generic;

    public class Message
    {
        public virtual Guid Id { get; set; }
        public virtual Guid? Schedule { get; set; }
        public virtual Guid? Owner { get; set; }
        public virtual string Sender { get; set; }
        public virtual IList<RecipientsResource> Recipients { get; set; }
        public virtual IList<ContactGroup> ContactGroups { get; set; }
        public virtual string Body { get; set; }
        public virtual string Subject { get; set; }
        public virtual DateTime? Time { get; set; }
        public virtual Guid? Wallet { get; set; }
        public virtual bool? Urgent { get; set; }
        public virtual string AckUri { get; set; }
        public virtual string ReplyUri { get; set; }
        public virtual string Ref { get; set; }
        public virtual string Folder { get; set; }
        public virtual string Status { get; set; }
        public virtual string Category { get; set; }
        public virtual string Type { get; set; }
        public virtual int? Validity { get; set; }
        public virtual DateTime? Created { get; set; }
        public virtual DateTime? Updated { get; set; }
        public virtual string AudioBase64 { get; set; }
        public virtual bool? Async { get; set; }
        public virtual string SpecificId { get; set; }
        public virtual Guid IdDomain { get; set; }
        public virtual string LargeAccount { get; set; }
        public virtual bool? Read { get; set; }

        public Message()
        {
            Recipients = new List<RecipientsResource>();
            ContactGroups = new List<ContactGroup>();
        }

        public Message(string content)
        {

        }

        public struct RecipientsResource
        {
            public string Value { get; set; }
            public string Status { get; set; }
            public DateTime? Updated { get; set; }
            public string Carrier { get; set; }
            public int? Duration { get; set; }
            public float? Price { get; set; }
            public int Attempts { get; set; }
            public DateTime? Time { get; set; }
        }

        public Message Clone()
        {
            return new Message
            {
                Id = Id,
                Schedule = Schedule,
                Owner = Owner,
                Sender = Sender,
                Recipients = new List<RecipientsResource>(Recipients),
                ContactGroups = new List<ContactGroup>(ContactGroups),
                Body = Body,
                Subject = Subject,
                Time = Time,
                Wallet = Wallet,
                Urgent = Urgent,
                AckUri = AckUri,
                ReplyUri = ReplyUri,
                Ref = Ref,
                Folder = Folder,
                Status = Status,
                Type = Type,
                Category = Category,
                Created = Created,
                Updated = Updated,
                Async = Async,
                SpecificId = SpecificId,
                IdDomain = IdDomain,
                AudioBase64 = AudioBase64,
                LargeAccount = LargeAccount,
                Read = Read
            };
        }
    }

}

File: src/clients/Java/src/main/java/br/com/takenet/takeio/entities/Message.java
package br.com.takenet.takeio.entities;

import java.util.Date;
import java.util.List;
import java.util.UUID;
import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
import com.fasterxml.jackson.annotation.JsonInclude;
import br.com.takenet.takeio.entities.adapters.DateAdapter;

//Never send empty properties in JSON
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Message {	
	private UUID id;
	private UUID schedule;
	private UUID owner;
	private String sender;
	private List<RecipientsResource> recipients;
	private List<ContactGroup> contactGroups;
	private String body;
	private String subject;
	@XmlJavaTypeAdapter(DateAdapter.class)
	private Date time;
	private UUID wallet;
	private Boolean urgent;
	private String AckUri;
	private String ReplyUri;
	private String Ref;
	private String Folder;
	private String Status;
	private String Category;
	private String type;
	private Integer validity;
	@XmlJavaTypeAdapter(DateAdapter.class)
	private Date created;
	private String updated;
	private String audioBase64;
	private Boolean async;
	private String specificId;
	private UUID idDomain;
	private String largeAccount;
	private Boolean read;
	
	public Message() {
	}
	
	public UUID getId() {
		return id;
	}
	public void setId(UUID id) {
		this.id = id;
	}
	public UUID getSchedule() {
		return schedule;
	}
	public void setSchedule(UUID schedule) {
		this.schedule = schedule;
	}
	public UUID getOwner() {
		return owner;
	}
	public void setOwner(UUID owner) {
		this.owner = owner;
	}
	public String getSender() {
		return sender;
	}
	public void setSender(String sender) {
		this.sender = sender;
	}
	public List<RecipientsResource> getRecipients() {
		return recipients;
	}
	public void setRecipients(List<RecipientsResource> recipients) {
		this.recipients = recipients;
	}
	public List<ContactGroup> getContactGroups() {
		return contactGroups;
	}
	public void setContactGroups(List<ContactGroup> contactGroups) {
		this.contactGroups = contactGroups;
	}
	public String getBody() {
		return body;
	}
	public void setBody(String body) {
		this.body = body;
	}
	public String getSubject() {
		return subject;
	}
	public void setSubject(String subject) {
		this.subject = subject;
	}
	public Date getTime() {
		return time;
	}
	public void setTime(Date time) {
		this.time = time;
	}
	public UUID getWallet() {
		return wallet;
	}
	public void setWallet(UUID wallet) {
		this.wallet = wallet;
	}
	public Boolean getUrgent() {
		return urgent;
	}
	public void setUrgent(Boolean urgent) {
		this.urgent = urgent;
	}
	public String getAckUri() {
		return AckUri;
	}
	public void setAckUri(String ackUri) {
		AckUri = ackUri;
	}
	public String getReplyUri() {
		return ReplyUri;
	}
	public void setReplyUri(String replyUri) {
		ReplyUri = replyUri;
	}
	public String getRef() {
		return Ref;
	}
	public void setRef(String ref) {
		Ref = ref;
	}
	public String getFolder() {
		return Folder;
	}
	public void setFolder(String folder) {
		Folder = folder;
	}
	public String getStatus() {
		return Status;
	}
	public void setStatus(String status) {
		Status = status;
	}
	public String getCategory() {
		return Category;
	}
	public void setCategory(String category) {
		Category = category;
	}
	public String getType() {
		return type;
	}
	public void setType(String type) {
		this.type = type;
	}
	public Integer getValidity() {
		return validity;
	}
	public void setValidity(Integer validity) {
		this.validity = validity;
	}
	public Date getCreated() {
		return created;
	}
	public void setCreated(Date created) {
		this.created = created;
	}
	public String getUpdated() {
		return updated;
	}
	public void setUpdated(String updated) {
		this.updated = updated;
	}
	public String getAudioBase64() {
		return audioBase64;
	}
	public void setAudioBase64(String audioBase64) {
		this.audioBase64 = audioBase64;
	}
	public Boolean getAsync() {
		return async;
	}
	public void setAsync(Boolean async) {
		this.async = async;
	}
	public String getSpecificId() {
		return specificId;
	}
	public void setSpecificId(String specificId) {
		this.specificId = specificId;
	}
	public UUID getIdDomain() {
		return idDomain;
	}
	public void setIdDomain(UUID idDomain) {
		this.idDomain = idDomain;
	}
	public String getLargeAccount() {
		return largeAccount;
	}
	public void setLargeAccount(String largeAccount) {
		this.largeAccount = largeAccount;
	}
	public Boolean getRead() {
		return read;
	}
	public void setRead(Boolean read) {
		this.read = read;
	}

}
